### PR TITLE
issue 647

### PR DESCRIPTION
**Pull request by olliethomas**
- wip
- make the soft predictions return a probability for each class

### CODE REVIEW

**Comment by olliethomas** [comment]
The idea for this PR is to use SoftPrediction more. For many classifiers we have the option to get a soft output, but we don't. So that was the idea. This led a little but down a rabbit hole as we're using "is the first dim >= 0.5" to determine the class label, when clearly argmax is much better for this. This also led to the Non-Binary test data being regenerated.

File: ethicml/algorithms/preprocess/calders.py [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
 from ranzen import implements
 
 from ethicml.utility import DataTuple, SoftPrediction, TestTuple, concat_dt
 
-from ..inprocess.logistic_regression import LRProb
 from .pre_algorithm import PreAlgorithm, T
 
 __all__ = ["Calders"]
 
+from ..inprocess.logistic_regression import LR
```
Comment (thomkeh) [inline ethicml/algorithms/preprocess/calders.py:14]: Should be above `__all__`, no?

File: ethicml/algorithms/preprocess/upsampler.py [hunk 2]
```diff
@@ -9,11 +9,12 @@
 
 from ethicml.utility import DataTuple, SoftPrediction, TestTuple
 
-from ..inprocess.logistic_regression import LRProb
 from .pre_algorithm import PreAlgorithm, T
 
 __all__ = ["Upsampler", "UpsampleStrategy"]
 
+from ..inprocess.logistic_regression import LR
```
Comment (thomkeh) [inline ethicml/algorithms/preprocess/upsampler.py:16]: Same

**Review by thomkeh — APPROVED** [review]
Makes sense that soft predictions can't be a pd.Series in the non-binary case.

**Pull request merged**